index.js

```diff
@@ -90,7 +90,8 @@ function setNestedProperty(object, property, value) {
         if (typeof property == "string" && property !== "") {
             var split = property.split(".");
             return split.reduce(function (obj, prop, idx) {
-                obj[prop] = obj[prop] || {};
+                const nextPropIsNumber = Number.isInteger(Number.parseInt(split[idx+1]))
```

Comment:
thanks for fixing! As explained in the conversation, Number.parseInt is error prone for a couple reasons. Could you update to `Number.isInteger(Number(split[idx + 1]))`?

Reply:
Sorry, maybe I was not see. But now I fixed the bug. Thank for passing me the tips, it was a great learning! I sent the new commit, now is just approve!